sql/connect/server/src/test/scala/org/apache/spark/sql/connect/pipelines/PipelineRefreshFunctionalSuite.scala

```diff
@@ -0,0 +1,367 @@
+      val pipeline = createTestPipeline(graphId)
+      registerPipelineDatasets(pipeline)
+
+      val startRun = PipelineCommand.StartRun
+        .newBuilder()
+        .setDataflowGraphId(graphId)
+        .setFullRefreshAll(true)
+        .addFullRefresh("a")
+        .build()
+
+      val exception = intercept[IllegalArgumentException] {
+        startPipelineAndWaitForCompletion(graphId, Some(startRun))
+      }
+      assert(
+        exception.getMessage.contains(
+          "Cannot specify a subset to refresh when full refresh all is set to true"))
+    }
+  }
+
+  test("validation: refresh and full_refresh cannot overlap") {
+    withRawBlockingStub { implicit stub =>
+      val graphId = createDataflowGraph
+      val pipeline = createTestPipeline(graphId)
+      registerPipelineDatasets(pipeline)
+
+      val startRun = PipelineCommand.StartRun
+        .newBuilder()
+        .setDataflowGraphId(graphId)
+        .addRefresh("a")
+        .addFullRefresh("a")
+        .build()
+
+      val exception = intercept[IllegalArgumentException] {
+        startPipelineAndWaitForCompletion(graphId, Some(startRun))
+      }
+      assert(
+        exception.getMessage.contains(
+          "Datasets specified for refresh and full refresh cannot overlap"))
+      assert(exception.getMessage.contains("a"))
+    }
+  }
+
+  test("validation: multiple overlapping tables in refresh and full_refresh not allowed") {
+    withRawBlockingStub { implicit stub =>
+      val graphId = createDataflowGraph
+      val pipeline = createTestPipeline(graphId)
+      registerPipelineDatasets(pipeline)
+
+      val startRun = PipelineCommand.StartRun
+        .newBuilder()
+        .setDataflowGraphId(graphId)
+        .addRefresh("a")
+        .addRefresh("b")
+        .addFullRefresh("a")
+        .build()
+
+      val exception = intercept[IllegalArgumentException] {
+        startPipelineAndWaitForCompletion(graphId, Some(startRun))
+      }
+      assert(
+        exception.getMessage.contains(
+          "Datasets specified for refresh and full refresh cannot overlap"))
+      assert(exception.getMessage.contains("a"))
+    }
+  }
+
+  test("validation: fully qualified table names in validation") {
+    withRawBlockingStub { implicit stub =>
+      val graphId = createDataflowGraph
+      val pipeline = createTestPipeline(graphId)
+      registerPipelineDatasets(pipeline)
+
+      val startRun = PipelineCommand.StartRun
+        .newBuilder()
+        .setDataflowGraphId(graphId)
+        .addRefresh("spark_catalog.default.a")
+        .addFullRefresh("a") // This should be treated as the same table
+        .build()
+
+      val exception = intercept[IllegalArgumentException] {
+        startPipelineAndWaitForCompletion(graphId, Some(startRun))
+      }
+      assert(
+        exception.getMessage.contains(
+          "Datasets specified for refresh and full refresh cannot overlap"))
+    }
+  }
+
+  private def verifyMultipleTableContent(
```

Comment:
Are we handling each table independently? If so, would it make sense to expose a function that operates on a single table and `foreach` over it where it's invoked?